### PR TITLE
#22 Detect go.mod in a directory without go files

### PR DESCRIPTION
**Pull request by aperezg**
Try to resolve the problems with the issue #22

### CODE REVIEW

**Comment by aperezg** [comment]
Mark, this resolve the case comment in the issue #22 but generate another problem, that I don't know how to solve on an elegant way.

If you run the command `pkger`, the binary file was be generated, but how any part of the code import this new package created by the tool itself, then the Apply method has never been called, and then the binary files has never used.

Do you have any idea that how to solve this case? Right now I don't see a clear solution, but I try to solve the problem

**Comment by markbates** [comment]
In the ticket you did go run x.go when you do that go doesn’t know about any other go file that isn’t included. Use go build to generate a binary or go run *.go will include all of the go files in that package.

File: here/info.go [hunk 1]
```diff
@@ -16,6 +16,16 @@ type Info struct {
 	Module     Module
 }
 
+// NewInfoFromPath initialize a Info with a basic information and his module
+// this method could be used when the Unmarshal information is not possible
+func NewInfoFromPath(path string, m Module) Info {
+	return Info{
+		Dir:        path,
+		ImportPath: "command-line-arguments",
```
Comment (markbates) [inline here/info.go]:
```suggestion
		ImportPath: "",
```

File: here/module.go [hunk 2]
```diff
@@ -5,11 +5,11 @@ import (
 )
 
 type Module struct {
-	Path      string `json:"path"`
-	Main      bool   `json:"main"`
-	Dir       string `json:"dir"`
-	GoMod     string `json:"go_mod"`
-	GoVersion string `json:"go_version"`
+	Path      string
```
Comment (markbates) [inline here/module.go]: Why were the json tags removed?

Comment (aperezg) [reply here/module.go]: the json output is:
```json
 "Module": {
                "Path": "github.com/friendsofgo/sample-pkgr",
                "Main": true,
                "Dir": "/Users/aperez/dev/go/src/github.com/friendsofgo/pkgr",
                "GoMod": "/Users/aperez/dev/go/src/github.com/friendsofgo/pkgr/go.mod",
                "GoVersion": "1.13"
        }
```
so the json tags are not necessary in this case

File: here/dir.go [hunk 3]
```diff
@@ -39,20 +40,30 @@ func Dir(p string) (Info, error) {
 				return i, err
 			}
 
+			if strings.Contains(es, "can't load package: package .") {
+				if _, err := os.Stat(fmt.Sprintf("%s/go.mod", p)); err == nil {
```
Comment (markbates) [inline here/dir.go]: This underlying issue isn't a missing `go.mod` file, but rather, not `.go` files at all and no `go.mod`. The test should check for the existence of `.go` files as well. Go requires at least one `.go` file *or* a `go.mod`.

Comment (aperezg) [reply here/dir.go]: Ok I will try to cover all the case around the Dir function itself, with tests.

**Comment by aperezg** [comment]
I've push the test, alse I did some modifications ^^' to share the code between the test_stuff you've created and the new test case.

**Pull request merged**